psycop/common/model_training_v2/loggers/disk_logger.py

```diff
@@ -0,0 +1,58 @@
+import json
+import logging
+from pathlib import Path
+from typing import Any
+
+from psycop.common.model_training_v2.config.baseline_registry import BaselineRegistry
+from psycop.common.model_training_v2.loggers.base_logger import BaselineLogger
+from psycop.common.model_training_v2.trainer.task.base_metric import CalculatedMetric
+
+
+@BaselineRegistry.loggers.register("disk_logger")
+class DiskLogger(BaselineLogger):
+    def __init__(self, experiment_path: str) -> None:
+        self.experiment_path = Path(experiment_path)
+        self.log_path = self.experiment_path / "logs.log"
+        self.cfg_log_path = self.experiment_path / "config.json"
+
+        self._l = self._setup_logging_module()
+
+    def info(self, message: str) -> None:
+        self._l.info(message)
+
+    def good(self, message: str) -> None:
+        self._l.info(message)
+
+    def warn(self, message: str) -> None:
+        self._l.warn(message)
+
+    def fail(self, message: str) -> None:
+        self._l.error(message)
+
+    def log_metric(self, metric: CalculatedMetric) -> None:
+        self._l.info(f"{metric.name}: {round(metric.value, 2)}")
+
+    def log_config(self, config: dict[str, Any]) -> None:
+        self.info(f"Logging config to json file at {self.cfg_log_path}")
+        with self.cfg_log_path.open("w") as f:
+            json.dump(config, f)
+
+    def _setup_logging_module(self) -> logging.Logger:
+        logger = logging.getLogger("DiskLogger")
+        logger.setLevel(logging.DEBUG)
+
+        self.log_path.parent.mkdir(exist_ok=True, parents=True)
+
+        # Create file handler
+        fh = logging.FileHandler(self.log_path, mode="a", encoding="utf-8")
+        formatter = logging.Formatter(
+            "%(asctime)s %(levelname)s %(name)s %(message)s",
+            datefmt="%Y-%m-%d %H:%M:%S",
+        )
+        fh.setFormatter(formatter)
+
+        # Add the handler to the logger
+        logger.addHandler(fh)
+
+        logger.info("Disk logger initialized")
+        return logger
```

Comment:
The logging module is so complicated, so bear with me:

As I understand, the logger is now in the namespace "GLOBAL.DiskLogger". Any new logger from logging that handles the GLOBAL namespace will also get log messages from this logger.

It don't think that's a problem, but just wanted to point to the complexity 👍